src/Aspire.Dashboard/Model/DashboardClient.cs

```diff
@@ -0,0 +1,355 @@
+                PooledConnectionIdleTimeout = TimeSpan.FromHours(2)
+            };
+
+            // https://learn.microsoft.com/aspnet/core/grpc/retries
+
+            var methodConfig = new MethodConfig
+            {
+                Names = { MethodName.Default },
+                RetryPolicy = new RetryPolicy
+                {
+                    MaxAttempts = 5,
+                    InitialBackoff = TimeSpan.FromSeconds(1),
+                    MaxBackoff = TimeSpan.FromSeconds(5),
+                    BackoffMultiplier = 1.5,
+                    RetryableStatusCodes = { StatusCode.Unavailable }
+                }
+            };
+
+            return GrpcChannel.ForAddress(
+                address,
+                channelOptions: new()
+                {
+                    HttpHandler = httpHandler,
+                    ServiceConfig = new() { MethodConfigs = { methodConfig } }
+                });
+        }
+    }
+
+    private void EnsureInitialized()
+    {
+        var priorState = Interlocked.CompareExchange(ref _state, StateInitialized, comparand: StateNone);
+
+        if (priorState is not StateNone)
+        {
+            ObjectDisposedException.ThrowIf(priorState is StateDisposed, this);
+            return;
+        }
+
+        _connection = Task.Run(() => ConnectAndStayConnectedAsync(_cts.Token), _cts.Token);
+
+        return;
+
+        async Task ConnectAndStayConnectedAsync(CancellationToken cancellationToken)
+        {
+            // Track the number of errors we've seen since the last successfully received message.
+            // As this number climbs, we hold off on 
+            var errorCount = 0;
+
+            await ConnectAsync().ConfigureAwait(false);
+
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                if (errorCount > 0)
+                {
+                    // The most recent attempt failed. There may be more than one failure.
+                    // We wait for a period of time determined by the number of errors,
+                    // where the time grows exponentially, until a threshold.
+                    var delay = ExponentialBackOff(errorCount, maxSeconds: 15);
+
+                    await Task.Delay(delay, cancellationToken).ConfigureAwait(false);
+                }
+
+                try
+                {
+                    await ProcessData().ConfigureAwait(false);
+                }
+                catch (RpcException ex)
+                {
+                    // Cancellation is reported in an RpcException, so unwrap it to throw the original OperationCanceledException instead.
+                    if (ex.StatusCode == StatusCode.Cancelled && ex.InnerException is OperationCanceledException oce)
+                    {
+                        // Rethrow cancellation
+                        ExceptionDispatchInfo.Capture(oce).Throw();
+                    }
```

Comment:
You can avoid this by setting ThrowOperationCanceledOnCancellation on the channel. (I really wanted OCE to be the default behavior but was unfortunately overruled)

https://learn.microsoft.com/en-us/aspnet/core/grpc/configuration?view=aspnetcore-8.0#configure-client-options